packages/core/database/lib/entity-manager/regular-relations.js

```diff
@@ -273,6 +281,96 @@ const cleanOrderColumns = async ({ id, attribute, db, inverseRelIds, transaction
   }
 };
 
+const cleanOrderColumnsForOldDatabases = async ({
+  id,
+  attribute,
+  db,
+  inverseRelIds,
+  transaction: trx,
+}) => {
+  const { joinTable } = attribute;
+  const { joinColumn, inverseJoinColumn, orderColumnName, inverseOrderColumnName } = joinTable;
+
+  const now = new Date().valueOf();
+
+  if (hasOrderColumn(attribute) && id) {
+    const tempOrderTableName = `tempOrderTableName_${now}`;
+    await db
+      .getConnection()
+      .raw(
+        `
+        CREATE TEMPORARY TABLE :tempOrderTableName:
+          SELECT
+            id,
+            (
+              SELECT count(*)
+              FROM :joinTableName: b
+              WHERE a.:orderColumnName: >= b.:orderColumnName: AND a.:joinColumnName: = b.:joinColumnName: AND a.:joinColumnName: = :id
+            ) AS src_order
+          FROM :joinTableName: a`,
+        {
+          tempOrderTableName,
+          joinTableName: joinTable.name,
+          orderColumnName,
+          joinColumnName: joinColumn.name,
+          id,
+        }
+      )
+      .transacting(trx);
+    await db
+      .getConnection()
+      .raw(
+        `UPDATE ?? as a, (SELECT * FROM ??) AS b
+          SET ?? = b.src_order
+          WHERE a.id = b.id`,
+        [joinTable.name, tempOrderTableName, orderColumnName]
+      )
+      .transacting(trx);
+    await db.getConnection().raw(`DROP TEMPORARY TABLE ${tempOrderTableName}`).transacting(trx);
```

Comment:
As the other promises can reject, maybe we want there to use a `try {...} finally { DROP }` to be sure to drop the temporary table ?

Reply:
Yes good idea. The temporary tables are not dropped when the transaction is rolled back